include/revng/Pipes/ModelInvalidationRule.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/ArrayRef.h"
+
+#include "revng/Pipeline/Container.h"
+#include "revng/Pipeline/Runner.h"
+#include "revng/Pipeline/Target.h"
+
+namespace revng::pipes {
+
+class ModelInvalidationRule {
+public:
+  ModelInvalidationRule() { getRulesRegistry().push_back(this); }
+
+  static llvm::ArrayRef<ModelInvalidationRule *> getRules() {
+    return getRulesRegistry();
+  }
+
+  virtual void registerInvalidations(const pipeline::ContainerBase &Container,
+                                     pipeline::TargetsList &ToRemove) const = 0;
+
+  static void
+  registerAllInvalidations(const pipeline::Runner &Runner,
+                           pipeline::Runner::InvalidationMap &Invalidations);
```

Comment:
This class has also a registry, but the `register*` methods do not register stuff into the registry.
What about `computeInvalidations`?

Reply:
why not just getInvalidations?